### ISSUE TITLE
Swiping through images in a gallery getting an error

### ISSUE THREAD

**Issue by AndreyKuleb** [dogfessional/react-swipeable#128]
**Describe the bug**
I use a [react-image-gallery](https://github.com/xiaolin/react-image-gallery), that use react-swipeable. When i am swiping gallery, an error in console appears (see image). There are the same [bug1](https://github.com/xiaolin/react-image-gallery/issues/321) and [bug2](https://github.com/xiaolin/react-image-gallery/issues/309) reports in react-image-gallery.

**Expected behavior**
In bug1 a decision was offered 

** Device/Browser **
MacBook Pro (Retina, 15-inch, Mid 2015)
MacOS Mojave 10.14.11
Google Chrome ver 72.0.3626.119 

**Additional context**
![2019-03-05 17 39 03](https://user-images.githubusercontent.com/20213742/53812970-9aa4fd80-3f6d-11e9-826a-78085626df70.png)
![2019-03-05 17 39 11](https://user-images.githubusercontent.com/20213742/53812974-9d9fee00-3f6d-11e9-812b-1d89ec6abdad.png)

**Comment by hartzis** [comment]
@AndreyKuleb thanks for creating the issue.

- Appears to be related to v4 `react-swipeable` and i have made a PR to that repo before to update the version.
- I think the best approach may be for `react-image-gallery` to call `e.preventDefault()` itself when it deems it appropriate
  - also possibly update to v5 😸 

I may be able to work on a PR for the above sometime over the next couple of weeks, but feel free to attempt if you'd like.

**PS** As a side note those errors are more like warnings, since `Calling preventDefault() for a non-cancelable event has no effect`, [MDN web docs - preventDefault#Notes](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault#Notes)

**Comment by StephanWagner** [comment]
Same problem here when I updated to 5.0.1

"react": "^16.8.4"
"react-swipeable": "^5.0.1"

Error message: [Intervention] Ignored attempt to cancel a touchmove event with cancelable=false, for example because scrolling is in progress and cannot be interrupted.

```
swipingNewsLeft(e, absX) {
  this.setState({newsSwiping: absX * -1});
}
...
<Swipeable
  onSwipingLeft={this.swipingNewsLeft}
  ...
>
```

**Comment by 13806** [comment]
Can confirm this appears when { preventDefaultTouchmoveEvent: true } in v5

**Comment by hartzis** [comment]
@13806 is your issue also related to https://github.com/xiaolin/react-image-gallery ? 

We just released [5.1.0](https://github.com/dogfessional/react-swipeable/releases/tag/v5.1.0) also addressed this issue a bit mainly for safari browsers, but should have helped with this warning.

**Comment by 13806** [comment]
@hartzis no, I'm using it standalone. Simple passing handlers into a div, like so:
```

  const handlers = useSwipeable({
    preventDefaultTouchmoveEvent: true,
    // ...
  })

 <div
     {...{
         ...handlers,
     }}
 >
     // ...
  </div>
```
When setting preventDefaultTouchmoveEvent to false, error does not occur.

**Comment by hartzis** [comment]
@13806 sorry to hear the 5.1.0 update broke touch.

It may be related to the new `ref` prop in the `handlers` object, https://github.com/dogfessional/react-swipeable#hook-details

In 5.1.0, we added a `ref` to the `useSwipeable` `handlers` and maybe you have another `ref` you're attaching to your div?

If you are also attaching a `ref` to the dom node that `...handlers` is being spread on you can update it like this:
```
function SwipeableUsingHook(props) {
  const handlers = useSwipeable(props)
  // your custom use for ref
  const customRef = ( el ) => {
    handlers.ref(el); // call the hanlders ref for useSwipeable to work correctly
    // do whatever custom stuff you need with el
    // ...
  }
  return (
    <div {...handlers} ref={customRef}>
      // ...
    </div>
  )
}
```

**Comment by 13806** [comment]
@13806 yes that seems to be it, I'm passing refs. However, your solution won't work for me since the main component needs to be able to take ref externally as a prop (and that other component passing down the ref doesn't know what handlers the useSwipeable main component would need, so I can't define handlers there). I wonder if there's a workaround.

**Comment by hartzis** [comment]
@13806 I'm really curious about how you're using `useSwipeable` and would love to help if I can.

Any way you can either past some code or create a [codesandbox.io](https://codesandbox.io/s/lrk6955l79?module=%2Fsrc%2FCarousel.js) example?

**Comment by hartzis** [comment]
PR created to address in `react-image-gallery` https://github.com/xiaolin/react-image-gallery/pull/399

**Comment by hartzis** [comment]
Closed for `react-swipeable` with #145
